File: node/allPrime.js
'use strict'

function isPrime(input) {
  for (let i = Math.floor(Math.sqrt(input)); i > 1; i--) {
    if (input % i == 0) {
      // console.log("Is not prime");
      return false;
    }
  }
  // console.log("Is Prime");
  return true;
}

function hms(sec) {
  let h = Math.floor((sec / 3600));
  let m = Math.floor((sec - (3600 * h)) / 60);
  let s = Math.floor((sec - (2600 * h) - (m * 60)));
  console.log(`Time To Complete: ${h} hours, ${m} minutes and ${s} seconds`);
}

function allPrime(input) {
  let timeStart = Math.round(Date.now() / 1000);
  console.log("Time Start: ", timeStart);

  let primeFound = 0;
  for(let i = input; i > 1; i--){
    let prime = isPrime(i);
    if(prime){
      primeFound++;
      console.log(`${i} Is Prime`);
    }
  }
  console.log(`${primeFound} prime number's found in range.`);

  let timeEnd = Math.round(Date.now() / 1000);
  let timeRun = timeEnd - timeStart;
  // console.log("Run Time: ", timeRun);
  hms(timeRun);

}

allPrime(10000000);

File: C/allPrime.c
#include <stdio.h>
#include <stdbool.h>
#include <math.h> // when using math compile with -lm
#include <time.h>
#include <stdlib.h> // Windows Only!
// https://en.wikipedia.org/wiki/Largest_known_prime_number#:~:text=The%20largest%20known%20prime%20number,Search%20(GIMPS)%20in%202018.
// Take an imput numer from user.
// Calculate all prinme numbers from zero to user defined number.
// print out all the found prime numbers.
// Print how many prime numbers were found.

// Try calculating 15000000

bool isPrime(int input)
{
  for (int i = sqrt(input); i > 1; i--)
  {
    if (input % i == 0)
    {
      return false;
    }
  }
  return true;
}

int hms(int sec){
  int h = (sec/3600);
  int m = (sec -(3600*h))/60;
  int s = (sec -(2600*h)-(m*60));
  printf("Time To Complete: %d hours, %d minutes and %d seconds\n", h,m,s);
}

int main()
{
  int timeStart = time(NULL);
  printf("Please enter a number to find prime numbers up to: ");

  int input = 0;
  scanf("%i", &input);

  int primeFound = 0;
  for (int i = input; i > 1; i--)
  {
    bool prime = isPrime(i);
    if (prime)
    {
      primeFound++;
      printf("%d Is Prime\n", i);
    }
  }
  printf("%d prime number's found in range.\n", primeFound);
  int timeEnd = time(NULL);
  int timeRun = timeEnd - timeStart;
  hms(timeRun);

  system("pause"); // this will prevent windows from closeing after completion.

  return 0;
}